server/broadcast.go

```diff
@@ -33,18 +33,33 @@ func (pi *packetInfo) broadcastMessage() {
 			pi.broadcastAll(msgs)
 		} else {
 			sendingPlayer := pi.tracker.playerByConnection(pi.conn)
+			if sendingPlayer == nil {
+				if debugMode {
+					fmt.Printf("[DirectMessage] Ignoring from %s because player no longer exists\n", getIP(pi.conn))
+				}
+				return
+			}
```

Comment:
changed logic check: consider direct message sender who drops as a significant error

Reply:
I wonder if this introduces a new race condition.  I mean a message can arrive then he can drop.. I would be very careful here.

Reply:
Assuming the ban logic works, then this shouldn't be an issue. The error does not mean that the player did something bad - just that it's time to clean up and disconnect this player/connection.

The ban logic works on blames (orthogonal to this issue) and the `isPassive` flag which should have been remove if the server was able to parse and get to this point of considering broadcasting the message.

The thing is we know the ban logic may be overactive at the moment so it is possible this is related to a bug.

Reply:
I think this logic change is unnecessary. We can do the debug message but I would not return. We still should forward the message and not change the behavior.

Reply:
I agree... in the simplest case, a player blames with a valid reason and drops because they have no need to hang around. This would keep the blamed player from getting a ban.

Reverted in ec5e15b